Mise en place du changement de langue avec ngx-translate

mamakh opened the pull request:
## Description

Design : PR #2
Mise en place de ngx-translate et du changement de langue en live.
Le portail a été passé sur ngx-translate, il faudra prévoir une migration d'i18n vers ngx-translate pour les autres applications.
Mise à jour des catégories.

Comparaison i18n et ngx-translate https://www.npmtrends.com/@ngx-translate/core-vs-angular-i18n

Nous avons ajouté la possibilité de changer de langue depuis le portail.
Des fichiers de langues ont été ajoutés dans ui-frontend-common et contiendront les textes sur les composants commun.
Ensuite pour chaque application(portal, identity, referential), des fichiers de langue supplémentaires seront ajoutés.
Des fichiers de langue pourront aussi être ajoutés dans le back pour pouvoir ajouter des éléments supplémentaires.


## Type de changement

-  Code et/ou Configuration : Nouvelle fonctionnalité
-  Build
-  Ansiblerie


## Tests

Tests effectués sur le portail.
Tests de changement de langue.

## Migration

Un script a été ajouté pour mettre le rôle update user à l'application portail afin qu'un utilisateur puisse changer de langue.
Cette langue est enregistrée dans les informations utilisateur pour que ça s'applique lors des prochaines connexions.

## Checklist

 [x] Mon code suit le style de code de ce projet.
 [x] J'ai commenté mon code, en particulier dans les classes et les méthodes difficile à comprendre.
 [x] J'ai rajouté les tests unitaires vérifiant mes fonctionnalités.
 [x ] J'ai rajouté les tests de non régression vérifiant mes fonctionnalités.
 [x] Les tests unitaires nouveaux et existants passent avec succès localement.
 [ ] Toutes les dépendances ont été mergées en priorité

## Contributeur

Xelians

Commit:
[US TRTL-157] component SelectLanguage & VitamTranslateService

[US TRTL-157] MultiLoadHttpLoader & translate templates

[US TRTL-157]add gulpfile to migrate from i18n to ngx-translate

[US TRTL-157] save/update language & sessionStorage

[US TRTL-157] sessionStorage & add ROLE_UPDATE_ME_USERS for portal & delete gulpfile.js

[US TRTL-157] clean i18n for portal in conf files

[US TRTL-157] review & fix ngx-translate version

Commit:
[US TRTL-427] add httpLoaderChildFactory to use local assets/i18n

fadil-zemmari-xelians committed:
[TRTL-484] Fix ngx translate with lazy modules

fadil-zemmari-xelians committed:
[US TRTL-473] Hide selection language button

fadil-zemmari-xelians committed:
[TECH] Fix ngx translate unit tests

Commit:
[TECH] Update categories for deployment & local configurations

Mocteezuma commented on ui/ui-frontend-common/package.json:
```diff
@@ -66,7 +70,11 @@
     "lodash-es": "^4.17.15",
-    "moment": "^2.26.0"
+    "moment": "^2.26.0",
+    "moment-mini": "^2.24.0",
```
Qu'apporte momment-mini exactement ?
Si c'est pour avoir les sources minifiées (et éviter d'avoir a embarquer la version non minifiée) est-il pertinant de laisser moment dans les dependencies ?

mamakh replied:
L'objectif c'est de virer moment et de garder moment-mini pour optimiser les bundles.
Le moment c'est un merge qui a foiré, on l'a mis sur un autre commit, je le mettrai sur cette branche

mamakh replied:
Commit poussé pour la correction sur la présence de moment-mini et moment.
Je me suis embrouillé sur le précédent message, on vire moment-mini et on garde moment avec la mise en place de moment-adapter

Mocteezuma reviewed (commented):
Une première review pour avancer sur le sujet en attendant la présentation de jeudi.

Est-ce qu'il faut qu'on prévoie des US d'adaptation de nos services (ingest / archive) pour adapter nos applications à ngx-translate ou est-ce que vous avez prévu de l'embarquer dans une de vos US futures ?
Si la question est encore en suspend a voir lors d'un prochain co-point.

mamakh commented:
Il faut que pour les prochains développements. les nouvelles traductions avec ngx-translate soit faites. 
Il y a de la charge de reprise pour les apps existantes, pour ça je ne sais pas trop qui va traiter quoi ou quelles apps.
Un sujet de discussion à avoir ...

Commit:
[TECH] PR review fix for moment-mini which was unused

pyXelians committed:
[Tech] Fix package-lock

mamakh committed:
Mise en place des analytics utilisateur et de la catégorie 'Dernières applications utilisées' (#69)

* [US TRTL-157] component SelectLanguage & VitamTranslateService

[US TRTL-157] MultiLoadHttpLoader & translate templates

[US TRTL-157]add gulpfile to migrate from i18n to ngx-translate

[US TRTL-157] save/update language & sessionStorage

[US TRTL-157] sessionStorage & add ROLE_UPDATE_ME_USERS for portal & delete gulpfile.js

[US TRTL-157] clean i18n for portal in conf files

[US TRTL-157] review & fix ngx-translate version

* [US TRTL-427] add httpLoaderChildFactory to use local assets/i18n

* [TRTL-484] Fix ngx translate with lazy modules

* [US TRTL-473] Hide selection language button

* [TECH] Fix ngx translate unit tests

* [TECH] Update categories for deployment & local configurations

* [TECH] Remove tar.gz assembly

* [TECH] PR review fix for moment-mini which was unused

* [US TRTL-170] : save last used applications

[US TRTL-170] : code review - rename applicationAnalytics class

[US TRTL-170] : tag application us last used only at the first visit

[US TRTL-170] : code review

* [US TRTL-172] : Get last used app back-end

[US TRTL-172] : Get last used app front-end
[US TRTL-172] : Fix import

[US TRTL-172] : Get last used apps update
[US TRTL-172]: Fix after conflict resolve
[US TRTL-172]: Fix last used app analytics loading

[US TRTL-172]: Fix duplicate when searching an application

[US TRTL-172]: Code review improvement

* [TECH] : Delete logger debug in test

* [TECH] Fix review - Create applicationIDConstant + userBuilder to test

* [Tech] Fix package-lock

* [TECH] FIX merge

* Choix du tenant par défaut & Sélection du tenant ou de l'organisation pour les apps multi-tenants et multi-organisations (#71)

* [US TRTL-172] : Get last used app back-end

[US TRTL-172] : Get last used app front-end
[US TRTL-172] : Fix import

[US TRTL-172] : Get last used apps update
[US TRTL-172]: Fix after conflict resolve
[US TRTL-172]: Fix last used app analytics loading

[US TRTL-172]: Fix duplicate when searching an application

[US TRTL-172]: Code review improvement

* [US TRTL-170] : save last used applications

[US TRTL-170] : code review - rename applicationAnalytics class

[US TRTL-170] : tag application us last used only at the first visit

[US TRTL-170] : code review

* [US TRTL-171] Persist last tenant identifier analytics

* [US TRTL-216] Add select tenant dialog on first user auth

[US TRTL-216] Fix last tenant identifier synchronization

[US TRTL-216] Improvement & add tenant by app filter

[US TRTL-216] Code improvement
[US TRTL-216] Add hierarchy app new portal components

[US TRTL-216] Fix select tenant component css
[US TRTL-169] Add select tenant inside app engine
[TECH] Add lastly used application translation

[US TRTL-169] Fix selected tenant display in app engine

[US TRTL-169] Code & architecture improvement

[US TRTL-169] Add translation & code review

[US-TRTL-216] Revert header features from identity

* [US TRTL-408] Add unknown app icon


[FIX TRTL-423] Fix tenant selection


[US TRTL-425] Add customer selection in new header


[TRTL-425] Code review

* [TECH] Apply Translate modifications in selecttenant & selectcustomer

* [TECH] Remove dev files for VITAM (these files are in .gitignore)

* [TECH] Fix TU

* [TECH] Delete ngOnDestroy not used

Co-authored-by: Fadil Zemmari <fadil.zemmari@xelians.fr>
Co-authored-by: Delphine <delphine.coursier@teamdlab.com>
Co-authored-by: Makhtar DIAGNE <makhtar.diagne@teamdlab.com>
Co-authored-by: pybelecalo <pierre-yves.belecalo@xelians.fr>

Co-authored-by: Delphine <delphine.coursier@teamdlab.com>
Co-authored-by: Mickael <mickael.varlet@xelians.fr>
Co-authored-by: Fadil <Fadil.zemmari@xelians.fr>
Co-authored-by: Makhtar DIAGNE <makhtar.diagne@teamdlab.com>
Co-authored-by: pybelecalo <pierre-yves.belecalo@xelians.fr>
